nncf/common/tensor_statistics/statistics.py

```diff
@@ -0,0 +1,53 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from abc import ABC
+
+
+class TensorStatistic(ABC):
+    pass
+
+
+class MinMaxTensorStatistic(TensorStatistic):
+    def __init__(self, min_values, max_values):
+        self.min_values = min_values
+        self.max_values = max_values
+
+    @classmethod
+    def from_stat(cls, statistic: TensorStatistic):
+        if isinstance(statistic, MinMaxTensorStatistic):
+            return cls(statistic.min_values, statistic.max_values)
+        if isinstance(statistic, MedianMADTensorStatistic):
```

Comment:
that implementation looks strange in `MinMaxTensorStatistic` class. 
it shouldn't know about other subclasses like `MedianMADTensorStatistic`

Reply:
Thanks for the comment. In this case, I reused the previous implementation. I guess that the main idea is that MinMaxTensorStatistic (as the basic one, which attributes are used for range initialization) can be created from another kind of TensorStatistic. For example, MinMaxTensorStatistic can be created from MedianMADTensorStatistic, which has different properties. That is why MinMaxTensorStatistic has to be aware of other TensorStatistic types and their properties.